File: search_js/bfs.js
var node = require('./utils').node;
var queue = [];
var visited = new Set();

function bfs(nroot) {
    queue.push(nroot);
    while (queue.length != 0) {
        current_node = queue.shift();
        console.log(current_node.name);
        visited.add(current_node);
        if (current_node.nodes.length != 0) {
            for (node of current_node.nodes) {
                if (!visited.has(node._id)) {
                    queue.push(node);
                }
            }
        }
    }
}

(function main() {
    rootNode = new node(0);
    rootNode.addChildren([new node(1), new node(2), new node(3)]);
    rootNode.nodes[0].addChildren([new node(4), new node(5)]);
    rootNode.nodes[1].addChildren([new node(6), new node(7)]);
    rootNode.nodes[2].addChildren([new node(8), new node(9)]);
    bfs(rootNode);
})();

File: search_py/bfs.py
from .utils import Node
from collections import deque

visited = set()
queue = deque()

def bfs(nroot):
    queue.appendleft(nroot)
    while queue:
        current_node = queue.pop()
        print(current_node.mesg)
        visited.add(current_node._id)
        if current_node.nodes:
            for node in current_node.nodes:
                if node._id not in visited:
                    queue.appendleft(node)


if __name__ == '__main__':
    root = Node(0)
    root.add_childs([Node(1), Node(2), Node(3)])
    root.nodes[0].add_childs([Node(4), Node(5)])
    root.nodes[1].add_childs([Node(6), Node(7)])
    root.nodes[2].add_childs([Node(8), Node(9)])
    bfs(root)